libcodechecker/analyze/log_parser.py

```diff
@@ -277,12 +276,31 @@ def parse_compile_commands_json(logfile, parseLogOptions):
                 if not os.path.isdir(inc_dir):
                     compile_opts[i] = '-I' + \
                         os.path.join(entry['directory'], inc_dir)
+
         action.analyzer_options = compile_opts
 
         action.lang = results.lang
         action.target = results.arch
         action.output = results.output
 
+        add_compiler_defaults = True
+
+        # With gcc-toolchain a non default compiler toolchain can be set.
+        # Clang will search for include paths and libraries based on the
+        # gcc-toolchain parameter.
+        # Detecting extra include paths from the host compiler could
+        # conflict with this.
+
+        # For example if the compiler in the compile command is clang
+        # and gcc-toolchain is set we will get the include paths
+        # for clang and not for the compiler set in gcc-toolchain.
+        # This can cause missing headers during the analysis.
+
+        for compiler_option in action.analyzer_options:
```

Comment:
We must add the resource includes of Clang with -isystem before adding any of the system headers of the original compiler.
And we do that in `extend_analyzer_cmd_with_resource_dir`. Could you please check if `--gcc-toolchain` is set it works well with those settings? I mean ,the files from Clangs' resource dir must be the first one on the include search path.